### PR TITLE
adds some javadoc, solves logger issue and  mockfactory issue and remove...

### PR DESCRIPTION
**Pull request by kardeslik**
...s exception handling for parser.

### CODE REVIEW

File: wasp/src/main/java/com/orhanobut/wasp/Logger.java [hunk 1]
```diff
@@ -43,15 +45,18 @@ private static void log(int logType, String message) {
         if (logLevel == LogLevel.NONE) {
             return;
         }
-        int length = message.length();
+        //get bytes of message with system's default charset (which is UTF-8 for Android)
+        final byte[] bytes = message.getBytes();
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/Logger.java]: final can be removed

File: wasp/src/main/java/com/orhanobut/wasp/parsers/Parser.java [hunk 2]
```diff
@@ -7,7 +9,7 @@
  */
 public interface Parser {
 
-    <T> T fromJson(String content, Type type) throws Exception;
+    <T> T fromJson(String content, Type type) throws JsonSyntaxException;
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/parsers/Parser.java]: JsonSyntaxException creates a coupling here. We should either introduce our own Json exception or it should stay Exception.

**Comment by kardeslik** [comment]
I have changed parser exception type to IOException.

**Comment by orhanobut** [comment]
LGTM

**Pull request merged**